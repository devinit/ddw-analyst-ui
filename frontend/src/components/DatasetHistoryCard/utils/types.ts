export interface SavedQueryData {
  id: number;
  saved_query_db_table: string;
  operation: number;
  full_query: string | null;
  active: boolean;
  description: string;
  created_on: string;
  user: string;
  status: string;
}
