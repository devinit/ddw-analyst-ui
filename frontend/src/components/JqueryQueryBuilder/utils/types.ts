export type JqueryQueryBuilderFilter = {
  condition?: string;
  rules?: (JqueryQueryBuilderFilterComparator | JqueryQueryBuilderFilter)[];
};

export type JqueryQueryBuilderHaving = {
  condition?: string;
  rules?: (JqueryQueryBuilderHavingComparator | JqueryQueryBuilderHaving)[];
};

export type JqueryQueryBuilderHavingComparator = {
  id: string;
  field: string;
  type: string;
  input: string;
  operator: JqueryQueryBuilderComps;
  value?: string | number | { plain: string | number } | { column: string; aggregate: string };
  values?: { value: string; label: string }[];
};

export type JqueryQueryBuilderFilterComparator = {
  id: string;
  field: string;
  type: string;
  input: string;
  operator: JqueryQueryBuilderComps;
  value: string | number;
};
export type JqueryQueryBuilderComps =
  | 'equal'
  | 'not_equal'
  | 'less'
  | 'less_or_equal'
  | 'greater'
  | 'greater_or_equal'
  | 'between'
  | 'in';

export type JqueryQueryBuilderFieldData = {
  id: string | number;
  label: string;
  type: string;
};

export type JqueryQueryBuilderIcons = {
  add_group: string;
  add_rule: string;
  remove_group: string;
  remove_rule: string;
  error: string;
};
