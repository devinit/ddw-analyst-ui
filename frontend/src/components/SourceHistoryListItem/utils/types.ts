export interface FrozenData {
  id: number;
  parent_db_table: string;
  frozen_db_table: string | null;
  active: boolean;
  description: string;
  created_on: string;
  user: string;
  status: string;
  logs: string | null;
}
