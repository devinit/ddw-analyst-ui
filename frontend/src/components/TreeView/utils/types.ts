import { ReactElement } from 'react';

export interface TreeViewProps {
  data: Data[];
  depth?: number;

  deleteElement?: ReactElement;

  getStyleClassCb?: (node: Data) => void;

  isCheckable?: (node: Data, depth?: number) => void;
  isDeletable?: (node: Data, depth?: number) => void;
  isExpandable?: (node: Data, depth?: number) => void;

  keywordChildren?: string;
  keywordChildrenLoading?: string;
  keywordKey?: string;
  keywordLabel?: string;

  loadingElement?: ReactElement;
  noChildrenAvailableMessage?: string;

  onCheckToggleCb?: (data: Data[], depth?: number) => void;
  onDeleteCb?: (node: Data, data: Data[], depth?: number) => void;
  onExpandToggleCb?: (node: Data, depth?: number) => void;
  onUpdateCb?: (data: unknown[], depth?: number) => void;

  transitionEnterTimeout?: number;
  transitionExitTimeout?: number;
}

export interface Data {
  id: number;
  name: string;
  isExpanded?: boolean;
  isChecked?: boolean;
  isChildrenLoading?: boolean;
  children?: Data[];
}
