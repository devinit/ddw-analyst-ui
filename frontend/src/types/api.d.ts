export interface APIResponse<T> {
  count: number;
  next: string;
  previous: string;
  results: T;
}
export interface Links {
  next: string | null;
  previous: string | null;
}
export type LinksMap = Map<keyof Links, Links[keyof Links]>;

export interface FetchOptions {
  limit?: number;
  offset?: number;
  link?: string;
  search?: string;
  id?: number;
}
