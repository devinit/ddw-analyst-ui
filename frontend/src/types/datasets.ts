import { List, Map } from 'immutable';

export interface Dataset {
  id: string;
  title: string;
  description: string;
  publication: string;
  releasedAt: string;
  geography: string;
  geographicCode: string | null;
  createdBy: string;
  licence: string | null;
  citation: string | null;
}
export type DatasetMap = Map<keyof Dataset, Dataset[keyof Dataset]>;
export type DatasetList = List<DatasetMap>;
