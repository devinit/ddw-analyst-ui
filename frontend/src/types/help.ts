export type MenuLink = {
  caption: string;
  url: string;
  disabled?: boolean;
  addDividerAfter?: boolean;
};
