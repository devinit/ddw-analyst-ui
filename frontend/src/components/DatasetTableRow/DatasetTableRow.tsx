import * as React from 'react';
import { Dataset } from '../../types/datasets';

interface ComponentProps {
  classNames?: string;
  dataset: Dataset;
}
export const DatasetTableRow: React.SFC<ComponentProps> = ({ children, classNames, dataset }) => {
  const renderActions = (): React.ReactNode =>
    React.Children.map(children, (child) => {
      if (React.isValidElement(child)) {
        return child;
      }
    });

  return (
    <tr className={classNames} data-testid="datasets-table-row">
      <td>{dataset.title}</td>
      <td>{dataset.publication}</td>
      <td>{new Date(dataset.releasedAt).toDateString()}</td>
      <td className="td-actions text-right">{renderActions()}</td>
    </tr>
  );
};

export { DatasetTableRow as default };
