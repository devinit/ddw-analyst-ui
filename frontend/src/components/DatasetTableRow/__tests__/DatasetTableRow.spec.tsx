/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { DatasetTableRow } from '../DatasetTableRow';
import { Dataset } from '../../../types/datasets';

const defaultDataset: Partial<Dataset> = {
  title: 'Dataset 1',
  publication: 'My first dataset',
  releasedAt: new Date('Tue Oct 08 2019').toDateString()
};

test('renders the default component correctly', () => {
  const renderer = TestRenderer.create(
    <DatasetTableRow dataset={ defaultDataset as Dataset } />
  ).toJSON();

  expect(renderer).toMatchSnapshot();
});

test('renders all component children as actions', () => {
  const actionContent = 'Actions Go Here!';
  const actionTestID = 'action-element';
  const tableBody = document.createElement('tbody');
  const { getByTestId } = render(
    <DatasetTableRow dataset={ defaultDataset as Dataset }>
      <div data-testid={ actionTestID }>{ actionContent }</div>
    </DatasetTableRow>,
    { container: document.body.appendChild(tableBody) }
  );
  const testElement = getByTestId(actionTestID);

  expect(testElement).toHaveTextContent(actionContent);
  expect(testElement.parentElement).toMatchInlineSnapshot(`
    <td
      class="td-actions text-right"
    >
      <div
        data-testid="action-element"
      >
        Actions Go Here!
      </div>
    </td>
  `);
});
