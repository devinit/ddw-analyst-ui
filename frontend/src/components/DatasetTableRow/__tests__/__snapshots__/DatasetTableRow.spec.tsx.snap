// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the default component correctly 1`] = `
<tr
  data-testid="datasets-table-row"
  onClick={[MockFunction]}
>
  <td>
    Dataset 1
  </td>
  <td>
    My first dataset
  </td>
  <td>
    Mon Oct 07 2019
  </td>
  <td
    className="text-right"
  />
</tr>
`;
